lib/ansible/modules/system/pamd.py

```diff
@@ -630,25 +630,83 @@ def add_module_arguments(self, rule_type, rule_control, rule_path, args_to_add):
         # Get a list of rules we want to change
         rules_to_find = self.get(rule_type, rule_control, rule_path)
 
-        changed = 0
+        changes = 0
 
         for current_rule in rules_to_find:
+            rule_changed = False
             if isinstance(args_to_add, str):
                 args_to_add = args_to_add.replace(" = ", "=")
                 args_to_add = args_to_add.split(' ')
             if not args_to_add:
                 args_to_add = []
-            # Create a list of new args that aren't already present
-            new_args = [arg for arg in args_to_add if arg not in current_rule.rule_args]
-            # If there aren't any new args to add, we'll move on to the next rule
-            if not new_args:
-                continue
 
-            current_rule.rule_args = current_rule.rule_args + new_args
+            # create some structures to evaluate the situation
+            new_args_d = dict()
+            no_eq_new_args = set()
+            for new_arg in args_to_add:
+                if new_arg.count("=") is 0:
+                    no_eq_new_args.add(new_arg)
+                else:
+                    pair = new_arg.split("=")
+                    new_args_d[pair[0]] = pair[1]
```

Comment:
Does this handle the case  where there may be multiple instances of `=` within an arg?

Reply:
This code was changed in a later commit.